Revit_Core_Engine/Create/View/ViewReflectedCeilingPlan.cs

```diff
@@ -39,66 +39,73 @@ public static partial class Create
+        [Input("viewDetailLevel", "Optional, the Detail Level of the view.")]
+        [Output("viewReflectedCeilingPlan", "The new view.")]
+        public static View ViewReflectedCeilingPlan(this Document document, Level level, string viewName = null, CurveLoop cropBox = null, ElementId viewTemplateId = null, ViewDetailLevel viewDetailLevel = ViewDetailLevel.Coarse)
         {
-            View result = null;
+            View newView = null;
 
-            ViewFamilyType vft = Query.ViewFamilyType(document, ViewFamily.CeilingPlan);
+            ViewFamilyType viewFamilyType = Query.ViewFamilyType(document, ViewFamily.CeilingPlan);
+
+            newView = Autodesk.Revit.DB.ViewPlan.Create(document, viewFamilyType.Id, level.Id);
+
+            Modify.SetViewDetailLevel(newView, viewDetailLevel);
 
-            result = Autodesk.Revit.DB.ViewPlan.Create(document, vft.Id, level.Id);
-            
-            Modify.SetViewDetailLevel(result, viewDetailLevel);
-            
             if (cropBox != null)
             {
                 try
                 {
-                    ViewCropRegionShapeManager vcrShapeMgr = result.GetCropRegionShapeManager();
-                    result.CropBoxVisible = true;
+                    ViewCropRegionShapeManager vcrShapeMgr = newView.GetCropRegionShapeManager();
+                    newView.CropBoxVisible = true;
                     vcrShapeMgr.SetCropShape(cropBox);
                 }
                 catch (Exception)
                 {
-                    BH.Engine.Base.Compute.RecordWarning("Could not create the Reflected Ceiling Plan with the provided crop box. Check if the crop box is a valid geometry and if the view's designated template accepts it to change.");
+                    BH.Engine.Base.Compute.RecordWarning("Could not create the Reflected Ceiling Plan with the provided crop box. Check if the crop box is a valid geometry.");
                 }
             }
 
             if (viewTemplateId != null)
             {
+                if (!(document.GetElement(viewTemplateId) as View).IsTemplate)
+                {
+                    BH.Engine.Base.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid View Template.");
+                    return newView;
+                }
+
                 try
                 {
-                    result.ViewTemplateId = viewTemplateId;
+                    newView.ViewTemplateId = viewTemplateId;
                 }
                 catch (Exception)
                 {
                     BH.Engine.Base.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid ElementId.");
                 }
             }
-            
+
             if (!string.IsNullOrEmpty(viewName))
             {
-                try
+                int number = 0;
```

Comment:
Shouldn't this entire section become a dedicated method, something around `SetViewName`? Assuming that the process looks the same for each view type.